src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -46,8 +47,29 @@ public override void RaiseEvent(OrchestrationContext unused, string eventName, s
             this.context.RaiseEvent(eventName, serializedEventData);
         }
 
+#if !FUNCTIONS_V1
+        public void ReportToAppInsights(OrchestrationRuntimeStatus status, string instanceId)

```

Comment:
Minor feedback on the naming:
* [AppInsights](https://www.appdirect.com/products/appinsights) is a different product not owned by Microsoft. Suggest to use either `AzureMonitor` or `ApplicationInsights`.
* This function itself doesn't do any specific thing regarding ApplicationInsights, it simply tags the current activity. Is is possible to have a name that reflects to what the function is actually doing? (e.g. TagCurrentActivity)?

Reply:
This is good feedback, thank you. I'll change it to `TagCurrentActivity` in my next commit

Reply:
As we've discussed last Friday, this code would only update the current activity, instead of emit extra event to ApplicationInsights. The actual code that emits data is probably somewhere in the Azure Functions host. If the intention is to be able to emit multiple events while the process is running, the host code has to be changed.

Reply:
Incorporated these changes in my newest commit